fail2ban/server/action.py

```diff
@@ -394,20 +394,16 @@ def substituteRecursiveTags(cls, tags):
 					# recursive definitions are bad
 					#logSys.log(5, 'recursion fail tag: %s value: %s' % (tag, value) )
 					return False
-				elif found_tag in cls._escapedTags:
-					# Escaped so won't match
+				if found_tag in cls._escapedTags or not tags.has_key(found_tag):
+					# Escaped or missing tags - just continue on searching after end of match
+					# Missing tags are ok - cInfo can contain aInfo elements like <HOST> and valid shell
+					# constructs like <STDIN>.
+					m = t.search(value, m.end())
```

Comment:
whatever, your example is anyway not correct, because the regular expression `r'<([^ >]+)>'` will prevent using "embedded" tags, like you specified...